examples/c/domains/src/publisher.c

```diff
@@ -90,31 +90,31 @@ int main(int argc, char** argv) {
     }
 
     // create publisher
-    iox2_port_factory_pub_sub_ref_h ref_service = iox2_cast_port_factory_pub_sub_ref_h(service);
+    iox2_port_factory_pub_sub_h_mut ref_service = iox2_cast_port_factory_pub_sub_h_mut(service);
     iox2_port_factory_publisher_builder_h publisher_builder =
         iox2_port_factory_pub_sub_publisher_builder(ref_service, NULL);
     iox2_publisher_h publisher = NULL;
     if (iox2_port_factory_publisher_builder_create(publisher_builder, NULL, &publisher) != IOX2_OK) {
         printf("Unable to create publisher!\n");
         goto drop_service;
     }
-    iox2_publisher_ref_h publisher_ref = iox2_cast_publisher_ref_h(publisher);
+    iox2_publisher_h_mut publisher_mut = iox2_cast_publisher_h_mut(publisher);
 
     int32_t counter = 0;
-    while (iox2_node_wait(node_ref_handle, 1, 0) == iox2_node_event_e_TICK) {
+    while (iox2_node_wait(node_handle_mut, 1, 0) == iox2_node_event_e_TICK) {
         counter += 1;
 
         // loan sample
         iox2_sample_mut_h sample = NULL;
-        if (iox2_publisher_loan(publisher_ref, NULL, &sample) != IOX2_OK) {
+        if (iox2_publisher_loan(publisher_mut, NULL, &sample) != IOX2_OK) {
             printf("Failed to loan sample\n");
             goto drop_publisher;
         }
-        iox2_sample_mut_ref_h sample_ref = iox2_cast_sample_mut_ref_h(sample);
+        iox2_sample_mut_h_mut sample_mut = iox2_cast_sample_mut_h_mut(sample);
```

Comment:
An unfortunate side effect of the changes. Do you have suggestions to improve this? Maybe using `_h_ref` instead of `_h_mut` for everything?